hazelcast/src/main/java/com/hazelcast/map/impl/QueryResult.java

```diff
@@ -30,29 +32,36 @@
 
 public class QueryResult implements DataSerializable {
 
-    private final Collection<QueryResultEntry> result;
+    private final Collection<QueryResultEntry> result = new LinkedHashSet<QueryResultEntry>();
 
     private Collection<Integer> partitionIds;
 
+    private transient long resultLimit;
+    private transient long resultSize;
+
     public QueryResult() {
```

Comment:
I try to go through 1 constructor. So I would call this(Long.MAX_VALUE). Having a single true constructor (and the rest convenience calls) reduces duplication (and bugs)